packages/database_v2/__WIP_pg_schema_review/artifacts/provision.sql

```diff
@@ -0,0 +1,222 @@
+
+-- CreateEnum
+CREATE TYPE "token_asset_status" AS ENUM ('Queued', 'Failed', 'Succeeded');
+
+-- CreateEnum
+CREATE TYPE "pin_status" AS ENUM ('PinFailed', 'Pinned', 'Pinning', 'PinQueued');
+
+-- CreateTable
+CREATE TABLE "block" (
+    "hash" TEXT NOT NULL,
+    "number" INTEGER NOT NULL,
+
+    PRIMARY KEY ("hash")
+);
+
+-- CreateTable
+CREATE TABLE "token" (
+    "id" TEXT NOT NULL,
+    "token_id" TEXT NOT NULL,
+    "mint_time" TIMESTAMP(3) NOT NULL,
+    "token_asset_id" TEXT NOT NULL,
+    "token_contract_id" TEXT NOT NULL,
+    "owner_id" TEXT NOT NULL,
+
+    PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "tokens_on_blocks" (
+    "block_hash" TEXT NOT NULL,
+    "token_id" TEXT NOT NULL,
+    "created" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY ("block_hash","token_id")
+);
+
+-- CreateTable
+CREATE TABLE "token_asset" (
+    "id" TEXT NOT NULL,
+    "token_uri" TEXT NOT NULL,
+    "ipnft" TEXT NOT NULL,
+    "problem" TEXT,
+
+    PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "metadata" (
+    "id" TEXT NOT NULL,
+    "cid" TEXT,
+    "source_id" TEXT NOT NULL,
+    "name" TEXT,
+    "description" TEXT,
+    "imageId" TEXT NOT NULL,
+    "image_uri" TEXT NOT NULL,
+
+    PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "resource" (
+    "status" "resource_status" NOT NULL,
+    "status_text" TEXT NOT NULL,
+    "uri" TEXT NOT NULL,
+    "ipfs_url" TEXT NOT NULL,
+    "problem" TEXT NOT NULL,
+    "cid" TEXT,
+    "created" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "updated" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY ("uri")
+);
+
+-- CreateTable
+CREATE TABLE "resources_on_metadata" (
+    "metadata_id" TEXT NOT NULL,
+    "resource_uri" TEXT NOT NULL,
+    "created" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY ("metadata_id","resource_uri")
+);
+
+-- CreateTable
+CREATE TABLE "content" (
+    "cid" TEXT NOT NULL,
+    "dag_size" INTEGER,
+    "created" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY ("cid")
+);
+
+-- CreateTable
+CREATE TABLE "pin" (
+    "id" TEXT NOT NULL,
+    "cid" TEXT NOT NULL,
+    "location_id" TEXT NOT NULL,
+    "pin" "pin_status" NOT NULL,
+    "status_text" TEXT,
+    "updated" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+    "created" TIMESTAMP(3) NOT NULL DEFAULT CURRENT_TIMESTAMP,
+
+    PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "pin_location" (
+    "id" TEXT NOT NULL,
+    "peer_id" TEXT NOT NULL,
+    "peer_name" TEXT NOT NULL,
+    "region" TEXT NOT NULL,
+
+    PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "token_contract" (
+    "id" TEXT NOT NULL,
+    "name" TEXT NOT NULL,
+    "symbol" TEXT NOT NULL,
+    "supports_eip721_metadata" BOOLEAN NOT NULL,
+
+    PRIMARY KEY ("id")
+);
+
+-- CreateTable
+CREATE TABLE "erc721_import_result" (
```

Comment:
I think we could probably rename this to `erc721_import`, I had result there to workaround with the fact that you have input types and output types in GraphQL and reflecting on this now all other output types end up without `result` suffix.

Reply:
yeah, that's a much better name. I think we're at the point where we're pas the the 1:1 from fauna and the schema is being iterated on and we can pick ideal names so I'm going to be making a pass thinking of that as well, looking for things like this